docs/dev/tutorial.md

```diff
@@ -1,125 +1,243 @@
+```go
+// SetFlags adds flags to the command. It is part of the Command interface in
+// the juju/cmd package.
+func (c *setQOTDAuthorCommand) SetFlags(f *gnuflag.FlagSet) {
+	// Collect the default output formatters.
+	formatters := make(map[string]cmd.Formatter, len(cmd.DefaultFormatters))
+	for k, v := range cmd.DefaultFormatters {
+		formatters[k] = v.Formatter
+	}
+	// Add the output related command flags and set the default formatter to
+	// "smart". This will automatically format strings for output.
+	c.out.AddFlags(f, "smart", formatters)
+}
+
+// Init initializes the command before running it. It collects the user supplied
+// arguments and throws an error if they are not as expected. It is part of the
+// Command interface in the juju/cmd package.
+func (c *setQOTDAuthorCommand) Init(args []string) error {
+	switch len(args) {
+	case 0:
+		return errors.Errorf("No quote author specified")
+	case 1:
+		c.author = args[0]
+		return nil
+	default:
+		//  CheckEmpty checks that there are no extra arguments.
+		return cmd.CheckEmpty(args[1:])
+	}
+}
+
+// Run executes the action of the command. It is part of the Command interface
+// in the juju/cmd package.
+func (c *setQOTDAuthorCommand) Run(ctx *cmd.Context) error {
+	// For now, just tell the user what they wrote.
+	return c.out.Write(ctx, "Quote author set to \""+c.author+"\"")
+}
+```
+
+There are two other methods that are part of the command interface:
+`IsSuperCommand` and `AllowIntersperedFlags`. We do not need to provide these.
+The `setQOTDAuthorComand` struct embeds the `modelcmd.ControllerCommandBase`
+which in several layers down embeds `cmd.Command` from the command package. This
+provides a default implementation.
+
+### Test the command
+
+Juju uses the go check library for testing. To test the command you will add a
+go check unit test suite.
+
+In the `qotd` directory, create a `package_test.go`, and hook up go check to
+work with the `go test` command:
+```go
+package qotd_test
+
+import (
+  stdtesting "testing"
+
+  gc "gopkg.in/check.v1"
+)
+
+func TestPackage(t *stdtesting.T) {
+  gc.TestingT(t)
+}
+```
+Next, in the `qotd` directory, create a `qotd_test.go` file and add a go check
+suite.
+
+```go
+package qotd_test
+
+import (
+  gc "gopkg.in/check.v1"
+
+  "github.com/juju/juju/internal/testing"
+)
+
+type SetQOTDAuthorSuite struct {
+  testing.BaseSuite
+}
+
+var _ = gc.Suite(&SetQOTDAuthorSuite{})
+```
+Embedding the `testing.BaseSuite` provides isolation for the test from the
+system it is running on.
+
+Now, add some tests of the command. We will test the command works as expected
+and returns the right errors.
+
+```go
+func (s *SetQOTDAuthorSuite) TestSetQOTDAuthor(c *gc.C) {
+  context, err := cmdtesting.RunCommand(c, qotd.NewSetQOTDAuthorCommand(), "Nelson Mandela")
+  c.Assert(err, jc.ErrorIsNil)
+  stdout := cmdtesting.Stdout(context)
+  c.Assert(stdout, gc.Equals, `Quote author set to "Nelson Mandela"\n`)
```

Comment:
this test failed for me, 
using c.Assert(stdout, gc.Equals, "Quote author set to \"Nelson Mandela\"\n") instead works for me

Reply:
Ah yea, thats my fault for changing the test after I put it in here! I've updated it as you suggested.